server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBlobsDAO.java

```diff
@@ -209,4 +215,25 @@ public BlobPart(BlobId blobId, int position, Optional<Row> row) {
             this.row = row;
         }
     }
+
+    @Override
+    public InputStream read(BlobId blobId) {
+        try {
+            Pipe pipe = Pipe.open();
+            ConsumerChainer<ByteBuffer> consumer = Throwing.consumer(
+                    bytes -> {
+                        pipe.sink().write(bytes);
+                        pipe.sink().close();
+                    }
+            );
+            readBytes(blobId)
+                    .thenApply(ByteBuffer::wrap)
+                    .thenAccept(
+                            consumer.sneakyThrow()
+                    );
+            return Channels.newInputStream(pipe.source());
+        } catch (IOException e) {
+            return null;
```

Comment:
no logging? BTW we try to not return null in our APIs.

Reply:
Should I throw a runtime instead of returning null ? 

The IOException in this case is supposedly raised by `Pipe#open()` which delegates to `SelectorProvider.provider().openPipe()` for which I could only find 2 implementation: 
one which throws an UnsupportedOperationException and one which never actually throws the Exception. 

Considering that this is supposed to be scaffolding for migrating to the inpustream based API I  didn't want to have extensive handling for the exception

Reply:
@mbaechler  
Do we really want to embed a checked IOException in the ObjectStore signature ?

I interpreted the test `readBytesShouldBeEmptyWhenNoExisting` as meaning that `IOExceptions` are handled at the ObjectStore level and not propagated through the code base. 

That leaves 2 options : 
- runtime exception+ error log
- empty inputstream + error log

Reply:
inputstream is not an option IMO, so let's do runtime exception, but we should then create a specific Exception for that and document it in the method signature

Reply:
alternative implementation
```
        CompletableFuture<byte[]> eventualBytes = readBytes(blobId);
        final PipedOutputStream out = new PipedOutputStream();
        eventualBytes.thenAccept(Throwing.<byte[]>consumer(out::write).sneakyThrow());
        try {
            return new PipedInputStream(out);
        } catch (IOException e) {
            LOGGER.error("can not occur in this specific case");
         throw new IllegalStateException("unable to create pipe",e);
        }
```

Reply:
actually 
```
        CompletableFuture<byte[]> eventualBytes = readBytes(blobId);
        final PipedOutputStream out = new PipedOutputStream();
        try {
            PipedInputStream pipedInputStream = new PipedInputStream(out);
            eventualBytes.thenAccept(Throwing.<byte[]>consumer(out::write).sneakyThrow());
            return pipedInputStream;
        } catch (IOException e) {
            LOGGER.error("can not occur in this specific case");
            throw new IllegalStateException("unable to create pipe",e);
        }
```
is slightly safer

Reply:
done